include/fiction/technology/sidb_nm_position.hpp

```diff
@@ -23,16 +23,18 @@ namespace fiction
  * @return A pair representing the `(x,y)` position of `c` in nanometers from the layout origin.
  */
 template <typename Lyt>
-constexpr std::pair<double, double> sidb_nm_position(const sidb_simulation_parameters& sp, const cell<Lyt>& c) noexcept
+constexpr std::pair<units::length::nanometer_t, units::length::nanometer_t>
+sidb_nm_position(const sidb_simulation_parameters& sp, const cell<Lyt>& c) noexcept
 {
     static_assert(has_siqad_coord_v<Lyt>, "Lyt is not based on SiQAD coordinates");
 
-    const auto x = static_cast<double>(c.x * sp.lat_a * 0.1);
-    const auto y = static_cast<double>(c.y * sp.lat_b * 0.1 + c.z * sp.lat_c * 0.1);
+    const auto x = units::length::nanometer_t(
+        units::convert<units::length::angstrom, units::length::nanometer>((c.x * sp.lat_a).value()));
+    const auto y = units::length::nanometer_t(
+        units::convert<units::length::angstrom, units::length::nanometer>((c.y * sp.lat_b + c.z * sp.lat_c).value()));
```

Comment:
Are the `.value()` calls needed?

Reply:
As far as I understand it, the convert function needs a double.